File: Frontend/friend-nest/src/app/services/post.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { AddPostRequest } from '../models/request/AddPostRequest';
import { Observable } from 'rxjs';
import { Post } from '../models/Post';
import { PostTo } from '../models/response/PostTo';
import { GetPostsRequest } from '../models/request/GetPostsRequest';
import { GetPostDetails } from '../models/request/GetPostDetails';
import { PostDetails } from '../models/response/PostDetails';

@Injectable({
  providedIn: 'root',
})
export class PostService {
  constructor(private http: HttpClient) {}

  addPost(payload: AddPostRequest): Observable<Post> {
    return this.http.post<Post>('http://localhost:8080/api/addPost', payload);
  }

  getPostsExceptUser(getPostsRequest: GetPostsRequest): Observable<PostTo[]> {
    return this.http.post<PostTo[]>(
      'http://localhost:8080/api/getPostsExceptUser',
      getPostsRequest,
    );
  }

  getPostsDetails(getPostDetails: GetPostDetails): Observable<PostDetails> {
    return this.http.post<PostDetails>(
      'http://localhost:8080/api/getPostDetails',
      getPostDetails,
    );
  }

  deletePost(postId: number): Observable<Post> {
    return this.http.delete<Post>(
      `http://localhost:8080/api/deletePost/${postId}`,
    );
  }
}

File: Backend/friendnest/src/main/java/pl/polsl/friendnest/service/PostService.java
package pl.polsl.friendnest.service;

import pl.polsl.friendnest.model.CommentTo;
import pl.polsl.friendnest.model.Post;
import pl.polsl.friendnest.model.PostTo;
import pl.polsl.friendnest.model.request.AddPostRequest;
import pl.polsl.friendnest.model.request.GetPostDetailsRequest;
import pl.polsl.friendnest.model.request.GetPostsRequest;
import pl.polsl.friendnest.model.request.QuotePostRequest;
import pl.polsl.friendnest.model.response.PostDetails;

import java.util.List;

public interface PostService {

    Post addPost(AddPostRequest payload);
    List<PostTo> getPostsToExceptUser(GetPostsRequest getPostsRequest);

    List<PostTo> getUserPosts(Long userId);
    List<CommentTo> getUserComments(Long userId);

    PostDetails getPostDetails(GetPostDetailsRequest getPostDetailsRequest);

    Post deletePost(Long postId);

    Post quotePost(QuotePostRequest quotePostRequest);
}
